doc/rfcs/0001-separate-component-version-in-cluster.md

```diff
@@ -0,0 +1,44 @@
+# Separate Component Version in Cluster
+
+## Summary
+
+Add a version field to each component on topology file. Allow user to upgrade single component to given version.
+
+## Motivation
+
+- New component TiDB Dashboard need to deployed with TiDB but not release with TiDB. Need a way to specify Dashboard version when deployment and upgrade.
```

Comment:
I suspect whether TiDB Dashboard is really capable of a standalone version.

This means it is possible that user deploy a TiDB Dashboard with a very old TiDB, for example, TiDB 4.0 with TiDB Dashboard master. I guess this will just make TiDB Dashboard not working at all.

Reply:
I think use different component for TiDB cluster should not be considered by TiUP PM. Because TiDB cannot ensure different versions can work well together. We should not do this if no one can guarantee the behaviors of a TiDB cluster containing the different versions of components.

Implementing this feature is very easy and the hard part is the different version compatibility.

Reply:
There comes already a demand that TiDB-Dashboard want's to release standalone versions different with TiDB's, so we as a framework need to be prepared with such usage cases.

I do agree we should only enable this config for certain components, at this time, maybe only dashboard, in the `Specification` definitions.

Reply:
> I think use different component for TiDB cluster should not be considered by TiUP PM. Because TiDB cannot ensure different versions can work well together. We should not do this if no one can guarantee the behaviors of a TiDB cluster containing the different versions of components.
> 
> Implementing this feature is very easy and the hard part is the different version compatibility.

tiup cluster is not only used in prod environment. I think we cloud only allow user to specify tidb-dashboard version by default. And add a flag like --allow-separate-vertion that allow user to specify version of any component to test or patch component.